scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/testmode/ResultsCollectingListener.scala

```diff
@@ -24,6 +24,10 @@ case class ResultsCollectingListener[T](holderClass: String, runId: TestRunId, v
 
   def results: TestResults[T] = ResultsCollectingListenerHolder.resultsForId(runId)
 
+  // Warning! close can't clean resources because listener is passed into each scenario subpart and it will be closed few times
+  // We have to use dedicated clean() method when we are sure that we consumed results instead.
+  override final def close(): Unit = {}
```

Comment:
why not call `clean` in `close` and have the previous implementation in `FlinkTestMain`? Is `close` used in other contexts that cant do `clean`?